docs/core/concepts/results.md

```diff
@@ -59,33 +50,53 @@ The most common scenario in which a user might need to directly interact with a
 <Result: 1>
 ```
 
-## Result Handlers
+## Persisting Results
+
+Subclasses of `Result` that align with a storage backend have a specific implementation of a `read` / `write` / `exists` interface for persisting its data. These can be used directly by a user to read, write, or check on arbitrary data in storage backends matching the result subclass. They are also used by the Prefect pipeline whenever checkpointing is turned on if it needs to retrieve results from persistent storage.
 
-Result handlers are a more public entity than the `Result` class, as they provide the means to persist the data from an in-memory `Result` object. A result handler is a specific implementation of a `read` / `write` interface for handling data. The only actual requirement for a result handler implementation is that the `write` method returns a JSON-compatible object. 
+Each of these methods depend on the `Result.location` attribute to determine where the data should actually be persisted. This `location` attribute is the only value serialized to the database of Prefect Core's server or Prefect Cloud. By default this location is autogenerated as a UUID, but it can be provided as a formattable string by the user to configure exactly where the result is located.
```

Comment:
It's actually a bit more than a UUID (and the implementation might differ slightly between types).  The reason this is important is that different Result types _might_ have to override the implementation of `default_location` (e.g., the `LocalResult` overrides this to support both Windows and Posix path types)

Reply:
More precise/backed away from specifying exactly in the docs in 8920e4eec